diff --git a/mk/os.Linux.mk b/mk/os.Linux.mk
index a8208c8..b4bd1ee 100644
--- a/mk/os.Linux.mk
+++ b/mk/os.Linux.mk
@@ -2,8 +2,8 @@
 #
 # Build recipes for GNU/Linux based operating systems.
 
-OS_DISTRIBUTION         != lsb_release -s -i || echo unknown
-OS_DISTRIBUTION_VERSION != lsb_release -s -r || echo unknown
+OS_DISTRIBUTION         != uname -s || echo unknown
+OS_DISTRIBUTION_VERSION != uname -r || echo unknown
 
 .if ${OS_DISTRIBUTION} == "unknown" || \
     ${OS_DISTRIBUTION_VERSION} == "unknown"
