diff --git a/src/lib/evas/meson.build b/src/lib/evas/meson.build
index 56dabb9..67d5ac9 100644
--- a/src/lib/evas/meson.build
+++ b/src/lib/evas/meson.build
@@ -6,7 +6,7 @@
 
 png = dependency('libpng')
 tiff = dependency('libtiff-4', required: get_option('evas-loaders-disabler').contains('tiff') == false)
-giflib = cc.find_library('gif')
+giflib = dependency('giflib', required: get_option('evas-loaders-disabler').contains('gif') == false)
 webp = dependency('libwebp', version: ['>=0.5.0'], required: get_option('evas-loaders-disabler').contains('webp') == false)
 webpdemux = dependency('libwebpdemux', version: ['>=0.5.0'], required: get_option('evas-loaders-disabler').contains('webp') == false)
 libopenjp2 = dependency('libopenjp2', required: get_option('evas-loaders-disabler').contains('jp2k') == false)
